openmdao/core/test/test_empty_jacobian.py

```diff
@@ -0,0 +1,128 @@
+""" Test to verify that components without outputs will work as expected as
+long as their linearize method returns an empty dict {}.
+"""
+
+
+__author__ = 'rfalck'
+
+import unittest
+
+from openmdao.api import Problem, Group, Component, ExecComp, IndepVarComp
+from openmdao.drivers.scipy_optimizer import ScipyOptimizer
+
+
+class InputComp(Component):
+
+    def __init__(self):
+
+        super(InputComp,self).__init__()
+
+        self.add_param(name='A',val=0.0)
+        self.add_param(name='B',val=0.0)
+
+    def solve_nonlinear(self, params, unknowns, resids):
+        pass
+
+    def linearize(self, params, unknowns, resids):
+        return {}
+
+
+class InputComp2(Component):
+
+    def __init__(self):
+
+        super(InputComp2,self).__init__()
+
+        self.add_param(name='A',val=0.0)
+        self.add_param(name='B',val=0.0)
+
+    def solve_nonlinear(self, params, unknowns, resids):
+        pass
+
+    def linearize(self, params, unknowns, resids):
+        return None
+
+
```

Comment:
I tested for a ValueError, but I did not test against the text of the error, since I thought that could change at some point without changing the behavior otherwise.  Even though the behavior should be equivalent to returning None, I defined a third test where the linearize method is not defined.